### PR TITLE
feat: add optional Language filter to ContentSearchQuery

### LINKED ISSUE #150: Add Language to ContentSearchQuery
## Description
Add an optional Language parameter to ContentSearchQuery to enable language-aware discovery and resolution across the CSV pipeline. Input should be case-insensitive and normalized internally to uppercase to match CSV schema language codes.

## Requirements
- Add string? Language to ContentSearchQuery (backward compatible).
- Normalize input case to uppercase before use in provider/resolver.
- Accept "All", "EN", "DE", "FR", "ES", "IT", "KO", "PL", "PT-BR", "ZH-CN", "ZH-TW".
- Update XML docs and API docs.
- Add unit tests for normalization and null handling.

## Acceptance Criteria
- [ ] ContentSearchQuery exposes Language.
- [ ] Language handled case-insensitively; normalized to uppercase.
- [ ] Docs updated and tests added.

## Technical Details
- Location: GenHub/Features/Content/Models/ContentSearchQuery.cs (or equivalent).
- Consumers: CSVContentProvider (#141), CSVDiscoverer (#139), CSVResolver (#140), GameInstallationValidator (#145).
- Example:
```csharp
/// <summary>Optional language filter used by CSV content pipeline.</summary>
/// <remarks>
/// Accepts case-insensitive input and is normalized to uppercase to match CSV schema values.
/// Values: "All", "EN", "DE", "FR", "ES", "IT", "KO", "PL", "PT-BR", "ZH-CN", "ZH-TW".
/// </remarks>
public string? Language { get; set; }
```

## Related
- #141 
- #139 
- #140 
- #145
- #143 
- #144

### PR DESCRIPTION
**Pull request by mnoserat**
## Summary
This PR adds support for an optional Language filter to ContentSearchQuery, enabling language-aware content discovery in the CSV pipeline.

## Issue
Closes #150

### CODE REVIEW

File: GenHub/GenHub.Core/Models/Content/ContentSearchQuery.cs [hunk 1]
```diff
@@ -78,4 +78,47 @@ public class ContentSearchQuery
...
+        normalized = normalized.ToUpperInvariant();
+
+        // Special case for "ALL" to "All"
+        if (normalized == "ALL")
+        {
+            return "All";
+        }
+
+        // Map variants
+        return normalized switch
```
Comment (Skyaero42) [inline GenHub/GenHub.Core/Models/Content/ContentSearchQuery.cs]: Where are all the other variants?

**Review by Skyaero42 — COMMENTED** [review]
Your commit and PR message and description dont match the code change.

Maybe validate AI generated content before putting it up for review?

**Comment by Skyaero42** [comment]
> > Your commit and PR message and description dont match the code change.
> > Maybe validate AI generated content before putting it up for review?
> 
> > Your commit and PR message and description dont match the code change.
> > Maybe validate AI generated content before putting it up for review?
> 
> In fact, I talked with RS yesterday and asked him what to write in the commit message and the pull request since I’m new to these topics. He told me he didn’t know either, so I asked GPT to write them for me.
> 
> I apologize on any mistakes

A PR description explains in short what you have changed and why, to help reviewers and anyone in the future that needs to revisit the PR.

Using AI for that is fine, but it is important to proofread. AI has the tendency of writing commit descriptions too literal (it describes the code changes itself, rather than the functionality and  thought behind it). In your case, it also listed things that weren't touched (like unit tests).

File: GenHub/GenHub.Core/Models/Content/ContentSearchQuery.cs [hunk 2]
```diff
@@ -94,31 +94,32 @@ public string? Language
 
     private static string? NormalizeLanguage(string? language)
     {
-        if (string.IsNullOrWhiteSpace(language))
+        // set default to "ALL" if not specified
+        // create switch for supported languages Brazilian Chinese English French German Italian Korean Polish Spanish
+        switch (language?.Trim().ToUpper())
         {
-            return language;
```
Comment (Skyaero42) [inline GenHub/GenHub.Core/Models/Content/ContentSearchQuery.cs]: Using a dictionary here would be a cleaner and less error-prone approach

example:
```csharp
private static readonly Dictionary<string, string> LanguageMap =
    new(StringComparer.OrdinalIgnoreCase)
    {
        ["EN"] = "EN",
        ["DE"] = "DE",
        ["FR"] = "FR",
        ["PL"] = "PL",
        ["ES"] = "ES",
        ["IT"] = "IT",
        ["KO"] = "KO",
        ["BR"] = "BR",
        ["CN"] = "CN",
        ["ZH"] = "CN",
        ["ZH-CN"] = "CN"
    };

public string NormalizeLanguage(string? language)
{
    if (language == null)
        return "ALL";

    var key = language.Trim().ToUpperInvariant();

    return LanguageMap.TryGetValue(key, out var normalized)
        ? normalized
        : "ALL";
}
```
